packages/forklift-console-plugin/src/modules/Plans/views/details/tabs/VirtualMachines/modals/PlanVMsEditModal.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import { PlanEditAction } from 'src/modules/Plans/utils/types/PlanEditAction';
+import { PlanEditPage } from 'src/modules/Plans/views/edit/PlanEditPage';
+import { useModal } from 'src/modules/Providers/modals';
+import { useInventoryVms } from 'src/modules/Providers/views';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import {
+  NetworkMapModelGroupVersionKind,
+  ProviderModelGroupVersionKind,
+  StorageMapModelGroupVersionKind,
+  V1beta1NetworkMap,
+  V1beta1Plan,
+  V1beta1Provider,
+  V1beta1StorageMap,
+} from '@kubev2v/types';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { Modal, ModalVariant } from '@patternfly/react-core';
+
+import './PlanVMsDeleteModal.style.css';
+import './PlanVMsEditModal.style.css';
+
+export interface PlanVMsEditModalProps {
+  plan: V1beta1Plan;
+  editAction: PlanEditAction;
+}
+
+export const PlanVMsEditModal: React.FC<PlanVMsEditModalProps> = ({ plan, editAction }) => {
+  const { toggleModal } = useModal();
+  const { t } = useForkliftTranslation();
+  const projectName = plan?.metadata?.namespace;
+
+  // Retrieve k8s source provider
+  const [sourceProvider, sourceProviderLoaded, sourceProviderLoadError] =
+    useK8sWatchResource<V1beta1Provider>({
+      groupVersionKind: ProviderModelGroupVersionKind,
+      namespaced: true,
+      name: plan?.spec?.provider?.source?.name,
+      namespace: plan?.spec?.provider?.source?.namespace,
+    });
+
+  // Retrieve k8s target provider
+  const [targetProvider, targetProviderLoaded, targetProviderLoadError] =
+    useK8sWatchResource<V1beta1Provider>({
+      groupVersionKind: ProviderModelGroupVersionKind,
+      namespaced: true,
+      name: plan?.spec?.provider?.destination?.name,
+      namespace: plan?.spec?.provider?.destination?.namespace,
+    });
+
+  const [providers, providersLoaded, providersLoadError] = useK8sWatchResource<V1beta1Provider[]>({
+    groupVersionKind: ProviderModelGroupVersionKind,
+    namespaced: true,
+    isList: true,
+    namespace: plan?.metadata?.namespace,
+  });
+
+  // Retrieve all k8s Network Mappings
+  const [networkMaps, networkMapsLoaded, networkMapsError] = useK8sWatchResource<
+    V1beta1NetworkMap[]
+  >({
+    groupVersionKind: NetworkMapModelGroupVersionKind,
+    namespaced: true,
+    isList: true,
+    namespace: plan?.metadata?.namespace,
+  });
+
+  // Retrieve all k8s Storage Mappings
+  const [storageMaps, storageMapsLoaded, storageMapsError] = useK8sWatchResource<
+    V1beta1StorageMap[]
+  >({
+    groupVersionKind: StorageMapModelGroupVersionKind,
+    namespaced: true,
+    isList: true,
+    namespace: plan?.metadata?.namespace,
+  });
+
+  const [vmData] = useInventoryVms(
+    { provider: sourceProvider },
+    sourceProviderLoaded,
+    sourceProviderLoadError,
+  );
```

Comment:
consider exporting to a custom hook, include all network calls

Reply:
hm, not sure about that, I still need to forward the result objects of the network calls into the child component anyways